File: src/format_sheet.py
from google.oauth2 import service_account
from googleapiclient.discovery import build

SCOPES = ['https://www.googleapis.com/auth/spreadsheets']
SERVICE_ACCOUNT_FILE = 'credentials.json'

def format_sheet():
    try:
        # Authentication
        creds = service_account.Credentials.from_service_account_file(
                SERVICE_ACCOUNT_FILE, scopes=SCOPES)
        service = build('sheets', 'v4', credentials=creds)

        # The ID of your spreadsheet
        spreadsheet_id = '1wgF0TeTV6nghyjurKFVHiso86CbBrDb8_eqzp4vNaHk'

        # Lấy thông tin tất cả các sheet
        sheet_metadata = service.spreadsheets().get(spreadsheetId=spreadsheet_id).execute()
        sheets = sheet_metadata.get('sheets', '')

        requests = []
        for sheet in sheets:
            sheet_id = sheet.get("properties", {}).get("sheetId", 0)
            
            requests.append({
                "repeatCell": {
                    "range": {
                        "sheetId": sheet_id
                    },
                    "cell": {
                        "userEnteredFormat": {
                            "wrapStrategy": "WRAP"
                        }
                    },
                    "fields": "userEnteredFormat.wrapStrategy"
                }
            })
            requests.append({
                "autoResizeDimensions": {
                    "dimensions": {
                        "sheetId": sheet_id,
                        "dimension": "ROWS"
                    }
                }
            })

        body = {
            'requests': requests
        }

        # Execute the request
        response = service.spreadsheets().batchUpdate(
            spreadsheetId=spreadsheet_id,
            body=body).execute()

        print('Successfully formatted the sheet using Python!')
        
    except Exception as e:
        print(f"Error formatting sheet: {e}")

if __name__ == '__main__':
    format_sheet()

File: src/format_sheet.js
const { google } = require('googleapis');
const path = require('path');

async function formatSheet() {
  try {
    const auth = new google.auth.GoogleAuth({
      keyFile: path.join(__dirname, 'credentials.json'),
      scopes: ['https://www.googleapis.com/auth/spreadsheets'],
    });

    const sheets = google.sheets({ version: 'v4', auth });
    const spreadsheetId = '1wgF0TeTV6nghyjurKFVHiso86CbBrDb8_eqzp4vNaHk';

    // Lấy thông tin tất cả các sheet
    const sheetData = await sheets.spreadsheets.get({ spreadsheetId });
    const sheetList = sheetData.data.sheets;
    
    const requests = [];
    
    for (const sheet of sheetList) {
      const sheetId = sheet.properties.sheetId;
      
      requests.push({
        repeatCell: {
          range: {
            sheetId: sheetId,
          },
          cell: {
            userEnteredFormat: {
              wrapStrategy: 'WRAP',
            },
          },
          fields: 'userEnteredFormat.wrapStrategy',
        }
      });
      
      requests.push({
        autoResizeDimensions: {
          dimensions: {
            sheetId: sheetId,
            dimension: 'ROWS',
          },
        }
      });
    }

    const response = await sheets.spreadsheets.batchUpdate({
      spreadsheetId,
      resource: { requests },
    });

    console.log('Successfully formatted the sheet!');
  } catch (error) {
    console.error('Error formatting sheet:', error.message);
  }
}

formatSheet();
